powerauth-restful-security-javaee/src/main/java/io/getlime/security/powerauth/rest/api/jaxrs/provider/PowerAuthAuthenticationProvider.java

```diff
@@ -269,9 +270,9 @@ public PowerAuthApiAuthentication validateToken(String tokenHeader, List<PowerAu
         // Validate the header
         try {
             PowerAuthTokenHttpHeaderValidator.validate(header);
-        } catch (InvalidPowerAuthHttpHeaderException e) {
-            logger.warn(e.getMessage(), e);
-            throw new PowerAuthAuthenticationException(e.getMessage());
+        } catch (InvalidPowerAuthHttpHeaderException ex) {
+            logger.warn("Token validation failed, error: {}", ex.getMessage());
+            throw new PowerAuthAuthenticationException("POWER_AUTH_TOKEN_INVALID");
```

Comment:
Let's pass `ex` as a parameter to the new exception to keep the cause somewhere.